Sources/AEPOptimize/Optimize.swift

```diff
@@ -82,6 +94,23 @@ public class Optimize: NSObject, Extension {
         registerListener(type: EventType.genericIdentity,
                          source: EventSource.requestReset,
                          listener: processClearPropositions(event:))
+
+        // Register listener for handling internal personalization request complete events
+        registerListener(type: EventType.optimize,
+                         source: EventSource.contentComplete,
+                         listener: processUpdatePropositionsCompleted(event:))
+
+        // Handler function called for each queued event. If the queued event is a get propositions event, process it
+        // otherwise if it is an Edge event to update propositions, process it only if it is completed.
+        eventsQueue.setHandler { event -> Bool in
+            if event.isGetEvent {
+                self.processGetPropositions(event: event)
+            } else if event.type == EventType.edge {
```

Comment:
use your extension?
```
} else if event.isUpdateEvent {
```

Reply:
Nope, we are tracking the event-request content dispatched to edge extension (as a result of update propositions API call). The content complete response event will contain the edge event's ID for parent ID.